src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/ObjectSequence.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using System.Runtime.InteropServices;
+
+namespace System.Diagnostics.Metrics
+{
+
+
+    internal partial struct ObjectSequence1 : IEquatable<ObjectSequence1>, IObjectSequence
+    {
+        public object? Value1;
+
+        public ObjectSequence1(object? value1)
+        {
+            Value1 = value1;
+        }
+
+        public override int GetHashCode() => Value1?.GetHashCode() ?? 0;
+
+        public bool Equals(ObjectSequence1 other)
+        {
+            return (Value1 == null && other.Value1 == null) || (Value1 != null && Value1.Equals(other.Value1));
+        }
+
+        public override bool Equals(object? obj)
+        {
+            return obj is ObjectSequence1 && Equals((ObjectSequence1)obj);
+        }
+    }
+
+    internal partial struct ObjectSequence2 : IEquatable<ObjectSequence2>, IObjectSequence
+    {
+        public object? Value1;
+        public object? Value2;
+
+        public ObjectSequence2(object? value1, object? value2)
+        {
+            Value1 = value1;
+            Value2 = value2;
+        }
+
```

Comment:
~GetHashCode is missing.
Same on the others below.~

Ah, I see it's a partial struct and the implementation is in other files. Can you add a comment here to make this obvious?

Reply:
Sure : ) Fwiw this pattern of putting platform specific implementations in another file with partial is a pattern that pops up a good bit in the libraries code.